test/integration/adoption_application_reviews_test.rb

```diff
@@ -20,4 +20,16 @@ class AdoptionApplicationReviewsTest < ActionDispatch::IntegrationTest
 
     assert_response :redirect
   end
+
+  test "all expected elements of an application are shown" do
+    sign_in users(:user_two)
+
+    get '/adopter_applications'
+
+    assert_select 'a', AdopterApplication.first.dog.name
+    assert_select 'p', "Applicant: #{AdopterApplication.first.adopter_account.user.first_name}
```

Comment:
Looks good! We could clean it up a little by creating an instance var (e.g., @adopter = ...)for the AdopterApplication.first.adopter_account.user and then calling first_name and last_name on that in the assertions?